cdap-app-fabric/src/main/java/io/cdap/cdap/internal/provision/ProvisioningService.java

```diff
@@ -236,6 +239,12 @@ public ClusterStatus getClusterStatus(ProgramRunId programRunId, ProgramOptions
                                    RetryableProvisionException.class::isInstance);
   }
 
+  @Nullable
+  private VersionInfo getAppCDAPVersion(ProgramOptions programOptions) {
+    String version = programOptions.getArguments().getOption(Constants.APP_CDAP_VERSION);
+    return version == null ? null : new ProjectInfo.Version(version);
```

Comment:
in the ProvisionerContext interface this is allowed to be null, should this be returning null instead of defaulting?

Reply:
Yes, it returns null if version is null